cli/compose/loader/loader.go

```diff
@@ -358,10 +363,18 @@ func resolveEnvironment(serviceConfig *types.ServiceConfig, workingDir string) e
 
 	serviceConfig.Environment = environment
 
+	// env is prioritized over the file
+	for k := range serviceConfig.Environment {
+		v, ok := lookupEnv(k)
```

Comment:
I think this is missing a check to see if `v == ""`. If `v` is not the empty string we should not override from the client environment.